src/client/app/components/HeaderButtonsComponent.jsx

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { Link } from 'react-router';
+import { Button } from 'reactstrap';
+import MenuModalComponent from './MenuModalComponent';
+import { hasToken } from '../utils/token';
+
+/**
+ * React component that controls the buttons in the Header
+ */
+export default class HeaderButtonsComponent extends React.Component {
+	constructor(props) {
+		super(props);
+		this.handleLogOut = this.handleLogOut.bind(this);
+	}
+
+	handleLogOut() {
+		localStorage.removeItem('token');
+		this.forceUpdate();
+	}
+
+	render() {
+		const urlArr = window.location.href.split('/');
```

Comment:
What's going on here? This is pretty complex logic that should be moved into a different function (or functions). And do we really want to use the current url as a source of state?

Reply:
This is not part this PR, but from a PR a couple months ago where the header component was redone to have it figure out which navigation buttons to display, instead of having the options passed down through props.

Reply:
Fair enough. I think it should be changed though.